pkg/logger/logger.go

```diff
@@ -0,0 +1,289 @@
+var globalLoggers = map[string]Logger{}
+var globalLoggersLock = sync.RWMutex{}
+
+// NewLogger creates new Logger instance.
+func NewLogger(name string) Logger {
+	globalLoggersLock.Lock()
+	defer globalLoggersLock.Unlock()
+
+	logger, ok := globalLoggers[name]
+	if !ok {
+		newLogger := logrus.New()
+		dl := &daprLogger{
+			name:              name,
+			jsonFormatEnabled: defaultJSONOutput,
+			logger:            defaultLogger(newLogger, name),
+		}
+		dl.EnableJSONOutput(defaultJSONOutput)
+		globalLoggers[name] = dl
+		logger = globalLoggers[name]
+	}
+
+	return logger
+}
+
+func defaultLogger(logger *logrus.Logger, name string) *logrus.Entry {
+	return logger.WithFields(logrus.Fields{
+		logFieldScope: name,
+		logFieldType:  LogTypeLog,
+	})
+}
+
+func getLoggers() map[string]Logger {
+	globalLoggersLock.RLock()
+	defer globalLoggersLock.RUnlock()
+
+	l := map[string]Logger{}
+	for k, v := range globalLoggers {
+		l[k] = v
+	}
+
+	return l
+}
+
+// EnableJSONOutput enables JSON formatted output log
+func (l *daprLogger) EnableJSONOutput(enabled bool) {
+	var formatter logrus.Formatter
+
+	l.jsonFormatEnabled = enabled
+	fieldMap := logrus.FieldMap{
+		// If time field name is conflicted, logrus adds "fields." prefix.
+		// So rename to unused field @time to avoid the confliction.
+		logrus.FieldKeyTime:  "@time",
+		logrus.FieldKeyLevel: logFieldLevel,
+		logrus.FieldKeyMsg:   logFieldMessage,
+	}
+
+	if enabled {
+		hostname, _ := os.Hostname()
+		l.logger.Data = logrus.Fields{
+			logFieldScope:    l.logger.Data[logFieldScope],
+			logFieldInstance: hostname,
+			logFieldType:     LogTypeLog,
+			logFieldDaprVer:  version.Version(),
+		}
+		formatter = &logrus.JSONFormatter{
+			DisableTimestamp: true,
+			FieldMap:         fieldMap,
+		}
+	} else {
+		l.logger.Data = logrus.Fields{
+			logFieldScope: l.logger.Data[logFieldScope],
+			logFieldType:  LogTypeLog,
+		}
+		formatter = &logrus.TextFormatter{
+			DisableTimestamp: true,
+			FieldMap:         fieldMap,
+		}
+	}
+
+	l.logger.Logger.SetFormatter(formatter)
+}
+
+// SetDaprID sets dapr_id field in log. Default value is empty string
+func (l *daprLogger) SetDaprID(id string) {
+	if l.jsonFormatEnabled {
+		l.logger.Data[logFieldDaprID] = id
+	}
+}
+
+// SetOutputLevel sets log output level
+func (l *daprLogger) SetOutputLevel(outputLevel LogLevel) {
+	lvl, ok := daprLoggingLevelToLogrusLevel[outputLevel]
+	if !ok {
+		lvl = daprLoggingLevelToLogrusLevel[defaultOutputLevel]
+	}
+	l.logger.Logger.SetLevel(lvl)
+}
+
+// WithLogType specify the log_type field in log. Default value is LogTypeLog
+func (l *daprLogger) WithLogType(logType string) Logger {
+	return &daprLogger{
+		name:   l.name,
+		logger: l.logger.WithField(logFieldType, logType),
+	}
+}
+
+// Info logs a message at level Info.
+func (l *daprLogger) Info(args ...interface{}) {
+	if l.jsonFormatEnabled {
+		l.logger.Data[logFieldTimeStamp] = utils.ToISO8601DateTimeString(time.Now())
```

Comment:
All times should be UTC times.

Reply:
`ToISO8601DateTimeString()` first converts the time to UTC and format it to ISO8601 (`2020-01-02T15:04:05.123000Z`) Please look at utils_test.go. 

So you do not need to call `UTC()`.